modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -1021,6 +1023,17 @@ class Session implements ISession {
         observers.each { trace -> trace.onFlowCreate(this) }
     }
 
+    void notifyFilePublish(Path destination) {
+        observers.each { observer -> {
```

Comment:
Why `observer -> { .. } `. Think this creates a nested closure (?). Also better to use just a `for` to keep the same pattern with the other handers

Reply:
there pending comment here